server/handlers.go

```diff
@@ -86,46 +78,46 @@ func (svr *Server) handleGetOPKeccakCommitment(w http.ResponseWriter, r *http.Re
 	if err != nil {
 		return fmt.Errorf("failed to decode commitment %s: %w", rawCommitmentHex, err)
 	}
+	// We use certV0 arbitrarily here, as it isn't used. Keccak commitments are not versioned.
+	// TODO: We should probably create a new route for this which doesn't require a versionedCert.
+	versionedCert := commitments.NewEigenDAVersionedCert(commitment, commitments.CertV0)
 
-	return svr.handleGetShared(r.Context(), w, commitment, commitmentMeta)
+	return svr.handleGetShared(r.Context(), w, versionedCert, commitments.OptimismKeccakCommitmentMode)
 }
 
 // handleGetOPGenericCommitment handles the GET request for optimism generic commitments.
 func (svr *Server) handleGetOPGenericCommitment(w http.ResponseWriter, r *http.Request) error {
-	versionByte, err := parseVersionByte(w, r)
+	certVersion, err := parseCertVersion(w, r)
 	if err != nil {
 		return fmt.Errorf("error parsing version byte: %w", err)
 	}
-	commitmentMeta := commitments.CommitmentMeta{
-		Mode:    commitments.OptimismGeneric,
-		Version: commitments.EigenDACommitmentType(versionByte),
-	}
-
-	rawCommitmentHex, ok := mux.Vars(r)[routingVarNamePayloadHex]
+	serializedCertHex, ok := mux.Vars(r)[routingVarNamePayloadHex]
 	if !ok {
 		return fmt.Errorf("commitment not found in path: %s", r.URL.Path)
 	}
-	commitment, err := hex.DecodeString(rawCommitmentHex)
+	commitment, err := hex.DecodeString(serializedCertHex)
 	if err != nil {
-		return fmt.Errorf("failed to decode commitment %s: %w", rawCommitmentHex, err)
+		return fmt.Errorf("failed to decode commitment %s: %w", serializedCertHex, err)
 	}
+	versionedCert := commitments.NewEigenDAVersionedCert(commitment, certVersion)
 
-	return svr.handleGetShared(r.Context(), w, commitment, commitmentMeta)
+	return svr.handleGetShared(r.Context(), w, versionedCert, commitments.OptimismGenericCommitmentMode)
 }
 
 func (svr *Server) handleGetShared(
 	ctx context.Context,
 	w http.ResponseWriter,
-	comm []byte,
-	meta commitments.CommitmentMeta,
+	versionedCert commitments.EigenDAVersionedCert,
+	mode commitments.CommitmentMode,
 ) error {
-	commitmentHex := hex.EncodeToString(comm)
-	svr.log.Info("Processing GET request", "commitment", commitmentHex, "commitmentMeta", meta)
-	input, err := svr.sm.Get(ctx, comm, meta)
+	commitmentHex := hex.EncodeToString(versionedCert.SerializedCert)
+	svr.log.Info("Processing GET request", "commitmentMode", mode,
+		"certVersion", versionedCert.CertVersion, "commitment", commitmentHex)
+	input, err := svr.sm.Get(ctx, versionedCert, mode)
 	if err != nil {
-		err = MetaError{
+		err = GETError{
```

Comment:
Did you intend to specify the `CertVersion` field for the `GETError`?

Reply:
[6a6cf14](https://github.com/Layr-Labs/eigenda-proxy/pull/379/commits/6a6cf1405f6f39f48ab4a25fb2fdfe7c5ca7b9c1)